### PR TITLE
ISPN-11620 Remove EncoderRegistry from DataConversion

### PR DESCRIPTION
**Pull request by gustavonalle**
https://issues.redhat.com/browse/ISPN-11620

### CODE REVIEW

File: core/src/main/java/org/infinispan/encoding/DataConversion.java [hunk 1]
```diff
@@ -143,30 +142,11 @@ private MediaType getStorageMediaType(Configuration configuration, boolean embed
       return MediaType.APPLICATION_UNKNOWN;
    }
 
-   public boolean isConversionSupported(MediaType mediaType) {
```
Comment (danberindei) [inline core/src/main/java/org/infinispan/encoding/DataConversion.java:146]: Can't remove these methods directly IMO, even if we don't expect applications to actually use them they're still part of the public API. Let's deprecate them, but simplify the implementation to simply delegate to `EncoderRegistry`.

Comment (gustavonalle) [reply core/src/main/java/org/infinispan/encoding/DataConversion.java:146]: The problem is to keep them the encoder registry in needed inside the data conversion, which defeats the whole purpose of the the PR :)

Comment (gustavonalle) [reply core/src/main/java/org/infinispan/encoding/DataConversion.java:146]: We can either: 
* deprecate them in the upcoming 10.1.6 and remove here
* just remove it here and write in the upgrade guide

Comment (danberindei) [reply core/src/main/java/org/infinispan/encoding/DataConversion.java:146]: IMO if we deprecate those methods and we never call them internally then it's safe to not inject the `EncoderRegistry` in those static instances (i.e. move the `encoderRegistry` assignment after the `if` line in `injectDependencies()`). E.g. https://github.com/infinispan/infinispan/commit/0ffe0955a415cef15e99ed06b9428c6c789b303e

Then everything works as it should, only stuff like `DataConversion.IDENTITY_KEY.convert(o, randomMediaType1, randomMediaType2)`, which shouldn't be a surprise for anyone.

I'm now working on ISPN-11629 and I'm thinking of deprecating those constants too, and using them only internally (if at all). 

OTOH I just noticed that `org.infinispan.encoding` and `org.infinispan.commons.encoding` are not `@public` packages. They are referenced in public APIs like `AdvancedCache`, so they should be in the javadoc. I don't think this gives a free pass to remove methods, but @tristantarrant might disagree :)

Comment (gustavonalle) [reply core/src/main/java/org/infinispan/encoding/DataConversion.java:146]: I'd personally deprecated in 10.1.7 and remove then altogether on 11

Comment (gustavonalle) [reply core/src/main/java/org/infinispan/encoding/DataConversion.java:146]: WDYT @danberindei?

Comment (danberindei) [reply core/src/main/java/org/infinispan/encoding/DataConversion.java:146]: @gustavonalle we can't deprecate in 10.1.x any more, so let's just deprecate it in 11.

Comment (gustavonalle) [reply core/src/main/java/org/infinispan/encoding/DataConversion.java:146]: @danberindei updated to simply deprecated it and removed the internal usages

File: core/src/main/java/org/infinispan/manager/EmbeddedCacheManager.java [hunk 2]
```diff
@@ -373,4 +374,10 @@ default EmbeddedCacheManagerAdmin administration() {
    }
 
    ClassWhiteList getClassWhiteList();
+
+   /**
+    * @since 11.0
+    * @return the {@link EncoderRegistry} associated with the cache manager.
+    */
+   EncoderRegistry getEncoderRegistry();
```
Comment (danberindei) [inline core/src/main/java/org/infinispan/manager/EmbeddedCacheManager.java:382]: Should this really be public API?

Comment (gustavonalle) [reply core/src/main/java/org/infinispan/manager/EmbeddedCacheManager.java:382]: What API do you suggest to expose that a conversion is supported and to convert data? ATM, at least internally, it grabs this EncoderRegistry from the registry every time

Comment (danberindei) [reply core/src/main/java/org/infinispan/manager/EmbeddedCacheManager.java:382]: Why would an application need to ask at runtime whether a conversion is supported? Internal components can keep injecting the `EncoderRegistry` from the global component registry.

Comment (gustavonalle) [reply core/src/main/java/org/infinispan/manager/EmbeddedCacheManager.java:382]: > Why would an application need to ask at runtime whether a conversion is supported? 

Those methods were in the DataConversion, and this PR deprecated them. Are you proposing to just deprecate them and provide no alternative?

Comment (danberindei) [reply core/src/main/java/org/infinispan/manager/EmbeddedCacheManager.java:382]: Yes, if someone actually needs an alternative we will provide it, but the deprecated methods will still work until the affected users realize they are affected and ask for an alternative.

Comment (gustavonalle) [reply core/src/main/java/org/infinispan/manager/EmbeddedCacheManager.java]: Ok, PR updated

**Comment by danberindei** [comment]
Merged, thanks Gustavo!

**Pull request merged**